cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -164,16 +169,25 @@ void publishToControlChannel(TetheringControlMessage message) throws Exception {
   }
 
   /**
-   * Use GZIPOutputStream to compress the LocalFile
+   * Use GZIPOutputStream to compress the LocalFile. First attempts to read from location factory; if the file does
+   * not exist in the location factory base path, attempts to read from local disk.
    */
   @VisibleForTesting
   byte[] getLocalFileAsCompressedBytes(LocalFile localFile) throws IOException {
-    File file = new File(localFile.getURI());
+    LOG.trace("Compressing local file with name '{}' and URI '{}'", localFile.getName(), localFile.getURI());
+    InputStream in;
+    try {
+      in = locationFactory.create(localFile.getURI()).getInputStream();
+    } catch (IllegalArgumentException e) {
+      // May not be of the expected location factory type. In this case, try reading from local disk.
+      File file = new File(localFile.getURI());
+      in = new FileInputStream(file);
+    }
     byte[] buffer = new byte[1024 * 500]; // use 500kb buffer
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    try (GZIPOutputStream os = new GZIPOutputStream(baos); FileInputStream fis = new FileInputStream(file)) {
```

Comment:
looks like the InputStream never gets closed anymore.

Reply:
Fixed